### PR TITLE
Migrate pages to new variation group structure

### PR DESCRIPTION
**Pull request by contolini**
This moves every component page to the new "group" format that we've been experimenting with over at https://github.com/cfpb-sandbox/design-system. It involves a heckuva lot of changes because the frontmatter of every markdown file has been slightly altered to accommodate the new format.

## Changes

- Replaces the `variations` field of every collection with a `variation_groups` field.
- Updates every markdown file to use the new `variation_groups` field.
- Replaces "Show/hide code & specs" button that was at the top of every page with a "Show/hide source" link at the bottom of each variation.

## Testing

1. Click around the [PR preview](https://deploy-preview-466--cfpb-design-system.netlify.app/design-system/components/) and look for obvious signs of wonkiness.
1. The browser tests have been updated and should pass: `yarn test:browser`

## Screenshots

| notifications page before | notifications page after |
|--------|-------|
| Screen Shot 2020-04-29 at 2 35 39 PM | Screen Shot 2020-04-29 at 2 35 51 PM |

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [x] Chrome on desktop
- [x] Firefox
- [x] Safari on macOS
- [x] Edge
- [x] Internet Explorer 9, 10, and 11
- [x] Safari on iOS
- [x] Chrome on Android

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: docs/assets/css/variation.less [hunk 1]
```diff
@@ -20,6 +17,35 @@
     }
 }
 
+.o-variation_group {
+    border-top: 1px solid @gray-40;
```
Comment (anselmbradford) [inline docs/assets/css/variation.less:21]: Maybe change these three lines into:

```
border: 1px solid @gray-40;
border-bottom: none;
```

File: docs/assets/css/variation.less [hunk 2]
```diff
@@ -20,6 +17,35 @@
...
+    &_description {
+        margin-bottom: 2rem;
+    }
+}
+
+.m-variation {
+    &_name {
+        margin-top: 2rem;
+    }
+    &:first-child {
```
Comment (anselmbradford) [inline docs/assets/css/variation.less:42]: Could get rid of a nest here with:

```
&:first-child .m-variation_name {
    margin-top: 0;
}
```

File: docs/assets/js/main.js [hunk 3]
```diff
@@ -24,24 +24,22 @@ if ( tabs ) {
...
-if ( toggleButton !== null ) {
-  toggleButton.addEventListener( 'click', ev => {
-    ev.preventDefault();
-    const codeIsHidden = toggleButton.getAttribute( 'data-code-hidden' );
-    if ( codeIsHidden ) {
-      showEls( codeSnippets );
-      toggleButton.removeAttribute( 'data-code-hidden' );
+// const toggleButton = document.querySelectorAll( 'a-toggle_code' );
+
+document.addEventListener( 'click', function( event ) {
```
Comment (anselmbradford) [inline docs/assets/js/main.js:29]: Consider moving the event manager to a named event handler.

```
document.addEventListener( 'click', handleDocumentClick );

/**
 * @param {MouseEvent} event - The mouse event object from the click.
 */
function handleDocumentClick( event ) { 
…
```

File: docs/assets/js/main.js [hunk 4]
```diff
@@ -24,24 +24,22 @@ if ( tabs ) {
...
+  if ( event.target.matches( '[data-toggle-code]' ) ) {
```
Comment (anselmbradford) [inline docs/assets/js/main.js:30]: Consider making this a guard clause:

```
if ( event.target.matches( '[data-toggle-code]' ) === false ) {
   let UNDEFINED;
   return UNDEFINED;
}

event.preventDefault();
…
```

File: docs/assets/js/main.js [hunk 5]
```diff
@@ -24,24 +24,22 @@ if ( tabs ) {
...
+    event.preventDefault();
+    const container = event.target.parentNode;
```
Comment (anselmbradford) [inline docs/assets/js/main.js:32]: Possibly move `event.target` to a variable at the top:

```
const target = event.target;
const container = target.parentNode;
const codeEl = document.querySelector( target.getAttribute( 'href' ) );
…

File: docs/assets/css/toggle-code-button.less [hunk 6]
```diff
@@ -1,19 +1,15 @@
...
-        .show-button { display: inline; }
-        .hide-button { display: none; }
+.a-toggle_code {
+    border-bottom: 1px solid @gray-40;
+    text-align: right;
+    margin: 0 -2rem;
+    padding: 0 2rem 1rem;
+    .a-btn {
+        text-transform: uppercase;
+        &.u-hidden {
```
Comment (anselmbradford) [inline docs/assets/css/toggle-code-button.less:10]: Why's this necessary? Isn't `u-hidden` already defined as `display:none`? Or does this not include cfpb-core?

File: docs/assets/css/tabs.less [hunk 7]
```diff
@@ -149,7 +151,9 @@
 }
 
 .variation-code-snippet-title,
-.variation-jinja-code-snippet-title {
+.variation-jinja-code-snippet-title,
+.variation-implementation-title,
+.variation-specs-title {
     .respond-to-min(641px, {
```
Comment (anselmbradford) [inline docs/assets/css/tabs.less:157]: TRIVIAL: space in parentheses used elsewhere.

File: docs/assets/css/tabs.less [hunk 8]
```diff
@@ -102,6 +97,13 @@
                     }
                 }
             }
+
+            &:first-child {
+                .govuk-tabs__tab {
```
Comment (anselmbradford) [inline docs/assets/css/tabs.less:102]: Hrm, customizing a third-party UI component smells bad.

File: docs/assets/css/toggle-code-button.less (hunk 6 above)
Comment (contolini) [reply docs/assets/css/toggle-code-button.less:10]: The `display: inline-block` on `.a-btn` overrides it. :/

Screen Shot 2020-05-01 at 1 06 06 PM

File: docs/assets/css/tabs.less (hunk 8 above)
Comment (contolini) [reply docs/assets/css/tabs.less:102]: Yeaaaaaahhhh all of `tabs.less` is overrides for the [gov.uk tabs](https://design-system.service.gov.uk/components/tabs/). It smells real bad. I'd like to migrate them into our own atomic component if it's a pattern we intend on using elsewhere.

File: docs/assets/css/toggle-code-button.less (hunk 6 above)
Comment (anselmbradford) [reply docs/assets/css/toggle-code-button.less:10]: wth that's not right

**Pull request merged**